WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomerSection/OrderCustomerSection.swift

```diff
@@ -0,0 +1,141 @@
+import SwiftUI
+import Networking
+
+/// Represents the Customer section
+///
+struct OrderCustomerSection: View {
+    let geometry: GeometryProxy
+
+    /// View model to drive the view content
+    @ObservedObject var viewModel: NewOrderViewModel
+
+    @State private var showAddressForm: Bool = false
+
+    var body: some View {
+        Group {
+            Divider()
+
+            VStack(alignment: .leading) {
+                HStack(alignment: .top) {
+                    Text(Localization.customer)
+                        .headlineStyle()
+
+                    Spacer()
+
+                    if viewModel.customerDataViewModel.isDataAvailable {
```

Comment:
This is a bit tricky, nested view model does not normally trigger updates, but in this case, they do because we are always mutating the whole `customerDataViewModel` 🤔 

I'm thinking that `customerDataViewModel ` is probably not a view model but just `CustomerData`? 😅

Reply:
But it's still a representation of view state, isn't it? 🙂
I think I understand your concerns, check f421be8afbc1e3330c0f0ccf2506cce562e85d3e for better usage of this model. I'm still thinking about navigation improvements on top of that.

Reply:
My conceptual problem with it is that since it's called a view model, anyone could go and add a `@Published` variable or code that updates `isDataAvailable` and at that point, the view model won't trigger updates because [nested view models do not trigger updates](https://rhonabwy.com/2021/02/13/nested-observable-objects-in-swiftui/) on `SwiftUI` 😓 

I think a good compromise if you want to keep the view model suffix is to extract that code into a private view and just observe/reference `customerDataViewModel` and not the outer view model.

What do you think?

Reply:
You can't add `@Published` wrapper because it's a struct. And you can't change its inner properties because all of them are non-mutable 🙂
My understanding is that viewmodel naming means special representation of data for the view. Like here - we don't mess with Addresses directly, instead extracting only some bits that wee need for the view.

> extract that code into a private view
What code do you mean? Here it's just plain static view created from struct input, no observation or dynamics. Struct changes - view recreated.

I want to get rid of additional parent big VM injection, but that's a navigation topic introduced and discussed in https://github.com/woocommerce/woocommerce-ios/pull/5757.

Reply:
> My understanding is that viewmodel naming means special representation of data for the view. Like here - we don't mess with Addresses directly, instead extracting only some bits that wee need for the view.

That is true! But in `SwiftUI` we also expect any code modified within a view model to trigger a view render pass, or is it just me? 😅 

I'm just worried about a new maintainer turning that view model into an observable object in the future and wondering why the view does not get updated.

Maybe I am over worrying!

> What code do you mean? Here it's just plain static view created from struct input, no observation or dynamics. Struct changes - view recreated.


I meant something like:

```swift
struct OrderCustomerSection: View {
     @ObservedObject var orderViewModel: NewOrderViewModel
     
     var body: some View {
          CustomerSubview1(VM: orderViewModel.customerDataViewModel)
          CustomerSubview2(VM: orderViewModel.customerDataViewModel)
     }
}

CustomerSubview1: View {

    let viewModel: NewOrderViewModel.CustomerDataViewModel

   var body: some View {
       if let fullName = viewModel.fullName {
             Text(fullName)
                 .bodyStyle()
      }
   }
}

```

That way if someone turns `customerDataViewModel` into an observed object, nothing else will break

Reply:
> That way if someone turns customerDataViewModel into an observed object, nothing else will break

That's a good safeguard, but turning plain struct into ObservableObject class with inner additional dynamics without parent view inspection - seems too reckless to me 😄
I agree with your code example - it looks clean and will work both ways. I'll reassess this code after #5757 (which may impact order viewmodel injection flow).

Thanks for sharing your idea and possible solution in so detailed way! 🙇‍♂️

Reply:
What do you think of removing intermediate vm completely and making view accept addressform vm and pass it as argument directly, without any observing?
Here is an example - 9e9e342c2367638cb36673a78104cdd22b059d30.
It seems to update data correctly.

Reply:
Seems to also work! 👍